core/src/main/scala/org/apache/spark/status/api/v1/StagesResource.scala

```diff
@@ -96,8 +96,9 @@ private[v1] class StagesResource extends BaseAppResource {
       @PathParam("stageAttemptId") stageAttemptId: Int,
       @DefaultValue("0") @QueryParam("offset") offset: Int,
       @DefaultValue("20") @QueryParam("length") length: Int,
-      @DefaultValue("ID") @QueryParam("sortBy") sortBy: TaskSorting): Seq[TaskData] = {
-    withUI(_.store.taskList(stageId, stageAttemptId, offset, length, sortBy))
+      @DefaultValue("ID") @QueryParam("sortBy") sortBy: TaskSorting,
+      @QueryParam("status") statuses: JList[TaskStatus]): Seq[TaskData] = {
```

Comment:
Why do we need to make it as array here? How about just `TaskStatus`?

Reply:
Ok, I'm also considering using `TaskStatus`. The reason I use JList is to make it consistent with stageList as below:

```
  @GET
  def stageList(@QueryParam("status") statuses: JList[StageStatus]): Seq[StageData] = {
    withUI(_.store.stageList(statuses))
  }
```

Reply:
Oh, sorry I was not aware of this. Please change it back to `JList` to make it consistent.